### PR TITLE
Add Minett-Wang bilinguals model

### PR DESCRIPTION
**Pull request by VicenteAR**
Add new file `lang_comp_bilinguals.py`. This file explores the _Minett-Wang bilinguals model_, where bilingual speakers enter as a new player to the competition.

### CODE REVIEW

File: abrams_strogatz/lang_comp_bilinguals.py [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+    fig = plt.figure()
+    ax = plt.axes()
+    plot = ax.matshow(popu, cmap=col, origin="lower", animated=True)
+    ax.xaxis.tick_bottom()
+    ax.xaxis.set_major_locator(plt.MultipleLocator(2))
+    ax.yaxis.set_major_locator(plt.MultipleLocator(2))
+    ax.xaxis.set_major_formatter(lambda val, pos: r"{}".format(int(val) + 1))
+    ax.yaxis.set_major_formatter(lambda val, pos: r"{}".format(int(val) + 1))
+    fig.colorbar(plot, ax=ax, ticks=colbar_tick, label="Language").ax.set_yticklabels(
+        ["B", "AB", "A"])
```
Comment (Guillemdb) [inline abrams_strogatz/lang_comp_bilinguals.py]: 66:  25 C812 missing trailing comma [flake8-commas]
```suggestion
        ["B", "AB", "A"],)
```

File: abrams_strogatz/lang_comp_bilinguals.py [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+    title = "Initial state" if (col == 0) else "Final state"
+    plot = ax.matshow(grid, cmap=color, origin="lower", animated=True)
+    ax.xaxis.tick_bottom()
+    ax.xaxis.set_major_locator(plt.MultipleLocator(2))
+    ax.yaxis.set_major_locator(plt.MultipleLocator(2))
+    ax.xaxis.set_major_formatter(lambda val, pos: r"{}".format(int(val) + 1))
+    ax.yaxis.set_major_formatter(lambda val, pos: r"{}".format(int(val) + 1))
+    ax.set_title(title)
+    fig.colorbar(
+        plot, ax=ax, ticks=colbar_tick, label="Language", fraction=0.047 * ratio
```
Comment (Guillemdb) [inline abrams_strogatz/lang_comp_bilinguals.py]: 261:  81 C815 missing trailing comma in Python 3.5+ [flake8-commas]
```suggestion
        plot, ax=ax, ticks=colbar_tick, label="Language", fraction=0.047 * ratio,
```

File: abrams_strogatz/lang_comp_bilinguals.py [hunk 3]
```diff
@@ -0,0 +1,300 @@
+import matplotlib as mpl
```
Comment (Guillemdb) [inline abrams_strogatz/lang_comp_bilinguals.py:2]: 1:   1 D100 Missing docstring in public module [flake8-docstrings]
```suggestion
"""This module implements the Minett-Wang bilingual model."""
import matplotlib as mpl
```

File: abrams_strogatz/lang_comp_bilinguals.py [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+            while (u < prob_change1) and (v < prob_change2):
+                u, v = np.random.uniform(), np.random.uniform()
+            popu[ii, jj] = 1 if (u < prob_change1) else -1
+
+    return popu
+
+
+def saving_process(popu: np.array, ph: int, key: bool = True):
+    """
+    Function used to save language grid plots.
```
Comment (Guillemdb) [inline abrams_strogatz/lang_comp_bilinguals.py]: 177:   1 D401 First line should be in imperative mood; try rephrasing [flake8-docstrings]
```suggestion
    Generate and save language grid plots.
```

File: abrams_strogatz/lang_comp_bilinguals.py [hunk 5]
```diff
@@ -0,0 +1,300 @@
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+import numpy as np
+
+# Initial condition. Randomly distributed speakers.
+
+
+def initial_cond(m: int, n: int, pa: float = 1./3., pb: float = 1./3.) -> np.array:
+    """Creates the initial (mxn) array.
```
Comment (Guillemdb) [inline abrams_strogatz/lang_comp_bilinguals.py]: 10:   1 D401 First line should be in imperative mood [flake8-docstrings]
```suggestion
    """Create the initial (mxn) array representing a population of speakers.
```

**Review by Guillemdb — CHANGES_REQUESTED** [review]

File: abrams_strogatz/lang_comp_bilinguals.py (hunk 5 above)
Comment (VicenteAR) [reply abrams_strogatz/lang_comp_bilinguals.py]: I have applied the requested changes and modified some docstrings.

File: abrams_strogatz/lang_comp_bilinguals.py [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+    result:
+
+    >>> print(periodic_boundary((6, 5), (5, 5)))
+    (1,0)
+    """
+    return tuple((i % s for i, s in zip(index, lattice_shape)))
+
+
+def language_dynamics(popu: np.array, m: int, n: int, s: float, a: float = 1.0) -> np.array:
+    """Population evolution. Change the language spoken by individuals.
```
Comment (Guillemdb) [inline abrams_strogatz/lang_comp_bilinguals.py]: 105:   1 D401 First line should be in imperative mood [flake8-docstrings]
```suggestion
    """
    Evolve the population of speakers following the Minett-Wang bilingual model.
```

**Review by Guillemdb — CHANGES_REQUESTED** [review]

File: abrams_strogatz/lang_comp_bilinguals.py (hunk 6 above)
Comment (VicenteAR) [reply abrams_strogatz/lang_comp_bilinguals.py]: Checked and corrected all function docstrings. Now, all first lines are in imperative mood.

**Pull request merged**